specs/APIReview_TextureStream.md

```diff
@@ -0,0 +1,705 @@
+  /// CreateTexture to allocate new texture.
+  /// The API also registers created shared handle to the browser once it
+  /// created the resource.
+
+  /// Unit for width and height is texture unit (in texels).
+  /// 'https://learn.microsoft.com/en-us/windows/win32/api/d3d11ns-d3d11-d3d11_texture2d_desc'
+  HRESULT CreateTexture(
+    [in] UINT32 widthInTexels,
+    [in] UINT32 heightInTexels,
+    [out, retval] ICoreWebView2StagingTexture** texture);
+  /// Returns reuseable texture for video frame rendering.
+  /// Once the renderer finishes rendering of texture's video frame, which
+  /// was requested by Present, the renderer informs the host so that it can
+  /// be reused. The host has to create new texture with CreateTexture
+  /// if the API return an error HRESULT_FROM_WIN32(ERROR_NO_MORE_ITEMS).
+  HRESULT GetAvailableTexture([out, retval] ICoreWebView2StagingTexture** texture);
+  /// Removes texture when the host removes the backed 2D texture.
+  /// The host can save the resources by deleting 2D textures
+  /// when it changes the frame sizes. The API will send a message
+  /// to the browser where it will remove texture.
+  HRESULT CloseTexture([in] ICoreWebView2StagingTexture* texture);
+  /// Adds the provided `ICoreWebView2Texture` to the video stream as the
+  /// next frame. The `ICoreWebView2Texture` must not be closed.
+  /// The `ICoreWebView2Texture` must have been obtained via a call to
+  /// `ICoreWebView2TextureStream::GetAvailableTexture` or `
+  /// ICoreWebView2TextureStream::CreateTexture` from this `ICoreWebView2TextureStream`.
+  /// If the `ICoreWebView2Texture` is closed or was created from a different
+  /// `ICoreWebView2TextureStream` this method will return `E_INVALIDARG`.
+  /// You should write your video frame data to the `ICoreWebView2Texture`
+  /// before calling this method.
+
+  /// After this method completes WebView2 will take some time asynchronously
+  /// to send the texture to the WebView2 processes to be added to the video stream.
+  /// Do not close or otherwise change the provided `ICoreWebView2Texture` after
+  /// calling this method. Doing so may result in the texture not being added to
+  /// the texture stream and the `ErrorReceived` event may be raised.
+  HRESULT PresentTexture([in] ICoreWebView2StagingTexture* texture)
+
+  /// Stops this texture stream from streaming and moves it into the stopped state.
+  /// When moving to the stopped state the `ICoreWebView2TextureStream Stopped`
+  /// event will be raised and in script the `MediaStreamTrack ended` event will
+  /// be raised. Once stopped, script may again call `Webview.getTextureStream`
+  /// moving the texture stream back to the start requested state.
+  /// See the `StartRequested` event for details.
+
+  /// Once stopped, calls to CreateTexture, GetAvailableTexture,
+  /// PresentTexture, and CloseTexture will fail with
+  /// COREWEBVIEW2_TEXTURE_STREAM_ERROR_NO_VIDEO_TRACK_STARTED.

```

Comment:
This seems wrong. This is an enum value not an HRESULT. What will the methods return when in the stopped state?

Reply:
It will return HRESULT_FROM_WIN32(ERROR_INVALID_STATE).